src/raftstore/store/store.rs

```diff
@@ -691,32 +692,28 @@ impl<T: Transport, C: PdClient> Store<T, C> {
         let t = SlowTimer::new();
         let pending_count = self.pending_raft_groups.len();
 
-        let mut ready_results: Vec<(u64, ReadyResult)> = Vec::with_capacity(pending_count);
-        for region_id in self.pending_raft_groups.drain() {
-            if let Some(peer) = self.region_peers.get_mut(&region_id) {
-                match peer.handle_raft_ready_append(&self.trans, &mut self.raft_metrics) {
-                    Err(e) => {
-                        // TODO: should we panic or shutdown the store?
-                        error!("{} handle raft ready append err: {:?}", peer.tag, e);
-                        continue;
-                    }
-                    Ok(Some(res)) => ready_results.push((region_id, res)),
-                    Ok(None) => {}
+        let (wb, append_res) = {
+            let mut ctx = ReadyContext::new(&mut self.raft_metrics, &self.trans, pending_count);
+            for region_id in self.pending_raft_groups.drain() {
+                if let Some(peer) = self.region_peers.get_mut(&region_id) {
+                    peer.handle_raft_ready_append(&mut ctx);
```

Comment:
if the writebatch count and data size exceed the max value, like count > 1024 or data size > 1m, we should write the batch.